includes/utilities.inc

```diff
@@ -21,23 +21,59 @@
  *   number. Each an array containing info.
  */
 function islandora_paged_content_get_pages(AbstractObject $object) {
-  $query = <<<EOQ
-PREFIX islandora-rels-ext: <http://islandora.ca/ontology/relsext#>
-SELECT ?pid ?page ?label ?width ?height
-FROM <#ri>
-WHERE {
-  ?pid <fedora-rels-ext:isMemberOf> <info:fedora/{$object->id}> ;
-       <fedora-model:label> ?label ;
-       islandora-rels-ext:isSequenceNumber ?page ;
-       <fedora-model:state> <fedora-model:Active> .
-  OPTIONAL {
-    ?pid <fedora-view:disseminates> ?dss .
-    ?dss <fedora-view:disseminationType> <info:fedora/*/JP2> ;
-         islandora-rels-ext:width ?width ;
-         islandora-rels-ext:height ?height .
- }
+  $pages = array();
+  // Try to use Solr first if enabled, or default to the RI if the Solr query
+  // fails for some reason.
+  if (variable_get('islandora_paged_content_use_solr_for_dimensions', FALSE)) {
```

Comment:
@jordandukart

Should have a module exists check.

Thanks!

----
William Panting
_Developer_
*[discoverygarden inc.](http://www.discoverygarden.ca) | Managing Digital Content*